`NcSelect`: Selected options should not overflow the container

susnux opened the pull request:
If the selected option is longer than the element it should not overflow the NcSelect but use ellipsis.

before | with `min-width: 0` | with hidden disabled search
---|---|---
![before](https://user-images.githubusercontent.com/1855448/219114982-eb4b2908-6928-44cf-bf53-f93e81d7f7e7.png)|![after](https://user-images.githubusercontent.com/1855448/219115009-1f9f168a-730a-48a6-b01a-a69f7f73b2ed.png)|![after-2](https://user-images.githubusercontent.com/1855448/219115020-f39775fc-a159-4dc8-9d58-0dcf454a4902.png)

As you can see currently the option overflows the select, note the actions (dropdown toggle) are missing because they are placed outside the select underneath other elements on the right.

After fixing the overflow there is still this weird height issues, which happens because `.vs__selected-options` have `flex-wrap` set and contain the search input even if it is disabled (`searchable=false`).
This can be fixed by removing the search input from the visible dom when not used.

I already submitted a upstream PR, but I doubt that it will be merged soon.

raimund-schluessler commented:
Fixing the width issue is fine, but the selected option still overflows:
![image](https://user-images.githubusercontent.com/2496460/219132478-65d7afdf-e303-4923-93f6-aea91ce1e444.png)

Seems like `.vs__selected` needs a width fix as well.

But, hiding the `input` field prevents the dropdown from closing. It now only closes when you select an option, but not when you click outside anymore. This is an upstream issue: https://github.com/sagalbot/vue-select/issues/1665
As a workaround for the Tasks app, I set `position: absolute;` for the input field:
https://github.com/nextcloud/tasks/blob/205f6c410e30d82e656f6e065c10562473ebc9af/src/components/AppSidebar/MultiselectItem.vue#L162
This doesn't hide it really, but moves it out of the flow.

raimund-schluessler reviewed (changes requested):
See https://github.com/nextcloud/nextcloud-vue/pull/3759#issuecomment-1431910791. This breaks closing the dropdown.

susnux commented:
> Fixing the width issue is fine, but the selected option still overflows :framed_picture: 

This is caused by `position: absolute` added when opening the menu.
I fixed this by setting a `max-width`.

> But, hiding the `input` field prevents the dropdown from closing. It now only closes when you select an option, but not when you click outside anymore.

Ah did not noticed that, thank you! Fixed that as well.
But I also added `visibility: hidden` to prevent it from being announced by screen readers.

susnux commented:
PR for the nextcloud fork of the vue-select: https://github.com/nextcloud-deps/vue-select/pull/18

(Forgot that we are using a custom fork and not the vanilla upstream)

raimund-schluessler commented on src/components/NcSelect/NcSelect.vue:
```diff
@@ -1007,6 +1007,36 @@ body {
+	 * There is an upstream pull request, if it is merged and released remove this fix
+	 * https://github.com/sagalbot/vue-select/pull/1756
+	 */
+	.vs__selected-options {
+		min-width: 0;
+		.vs__selected {
+			min-width: 0;
+		}
+	}
```
I just noticed that these two lines create issues when the `NcSelect` component has `:no-wrap="true"`set. See here:
![Screenshot 2023-02-22 at 12-45-19 Nextcloud Vue Style Guide](https://user-images.githubusercontent.com/2496460/220611641-f2789437-48d1-4850-b4ee-cde8dc4339bd.png)

raimund-schluessler replied:
@susnux I created a new issue for this: https://github.com/nextcloud/nextcloud-vue/issues/3808

mejo- replied:
Mh, shall we revert the PR for this release @raimund-schluessler?

raimund-schluessler replied:
I don't really have a preference here.

mejo- replied:
I don't think there's a need for rush, so let's try to get this solved before releasing 7.7.0. @susnux would you prefer to look for a fix or to revert the PR for 7.7.0?

susnux replied:
I will look for a fix.
